libraries/AP_RCProtocol/AP_RCProtocol.cpp

```diff
@@ -539,6 +547,38 @@ bool AP_RCProtocol::protocol_enabled(rcprotocol_t protocol) const
     return ((1U<<(uint8_t(protocol)+1)) & rc_protocols_mask) != 0;
 }
 
+void AP_RCProtocol::handle_radio_rc_channels(const mavlink_radio_rc_channels_t* packet)
+{
+    // receiving this message is also used to check if the receiver is present
+    // so let's first do the receiver detection
+    if (_detected_protocol == AP_RCProtocol::NONE) { // still searching
+#if AP_RC_CHANNEL_ENABLED
+        rc_protocols_mask = rc().enabled_protocols();
+#endif
+        if (!protocol_enabled(MAVLINK_RADIO)) return; // not our turn
+        _detected_protocol = AP_RCProtocol::MAVLINK_RADIO;
+    }
+
+    // here now comes the message handling itself
+
+    if (_detected_protocol != AP_RCProtocol::MAVLINK_RADIO) {
+        return;
+    }
+
+    // update the field, so that the backend can see it
+    memcpy(&(mavlink_radio.rc_channels), packet, sizeof(mavlink_radio_rc_channels_t));
```

Comment:
Can this be a structure assignment instead of memcpy?

Reply:
added virtual handlers to backend, which makes this field and line go altogether. Done.